### PR TITLE
e2e: add tests for organization settings cascade

### PR DESCRIPTION
**Pull request by unknwon**
Adds e2e tests for organization settings cascade, which covers ([source](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@57c1c6f9474d6929917b7de783a8e7ce3d8c6f45/-/blob/web/src/regression/org.test.ts#L81:1)):

- Create and delete organizations.
- Update an organization's settings and get cascaded settings.
- Remove the user from the organization no longer gets cascaded settings.

Easier to review by commit.

Part of #10068

### CODE REVIEW

File: dev/e2e/organization_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		}
+	}()
+
+	t.Run("settings cascade", func(t *testing.T) {
+		err := client.OverwriteSettings(orgID, `{"quicklinks":[{"name":"Test quicklink","url":"http://test-quicklink.local"}]}`)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		{
```
Comment (ryanslade) [inline dev/e2e/organization_test.go:32]: Why the extra bracket here?

File: dev/e2e/search_test.go [hunk 2]
```diff
@@ -4,9 +4,60 @@ package main
...
+	}
+	defer func() {
+		err := client.DeleteExternalService(esID)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	err = client.WaitForReposToBeCloned(
+		"github.com/sourcegraph/java-langserver",
```
Comment (ryanslade) [inline dev/e2e/search_test.go:47]: Maybe create a shared slice as it's used twice in this test

File: dev/e2e/organization_test.go (hunk 1 above)
Comment (unknwon) [reply dev/e2e/organization_test.go:32]: To intentionally set variable scope so the same variables names can be reused later (in another variable scope).

File: dev/e2e/search_test.go (hunk 2 above)
Comment (unknwon) [reply dev/e2e/search_test.go:47]: This one has "github.com/" prefix where the other one does not 🤦

**Pull request merged**